Clinical-Genomics/cg
# Mutacc PR is reviewed

What is left to do?:
- [x] wait for AR to test mutacc
- [ ] review PR
- [x] wait for AR to update servers 






This adds a mutacc-auto api to ```cg```. This PR makes cases uploadable to the mutacc db via the ```cg```CLI.

**How to test**:

 1. go to hasta
 2. source stage environment
 3. install this branch of cg on stage
 5. Update config files for cg to include configurations for mutacc by updating the stage servers repo with the feature branch 'mutacc'
 6. Make sure mutacc, and mutacc-auto are installed on S_mutacc
 7. run ```$ cg upload process-solved --case-id <case_id>```for some case that exists in stage
 8. run ```$ cg upload process-solved --days-ago <days>```to upload all cases in stage updated within <days> days ago (you might need to enter scout-stage and update one of the cases in scout)
 9. run ```$ cg upload processed-solved``` to upload the cases that has been processed by mutacc to the database.
 10. Check so that the cases has been added to database 
      - by sourcing S_mutacc
      - do mutacc db view cases
      - Check so that the reads have been extracted from the cases and are placed in the root directory of mutacc.   



**Expected outcome**:

 1. After Running ```$ cg upload process-solved --case-id <case_id>``` it should be logged that the reads are extracted for the specified case, where the temporary files of the case are stored for mutacc (These should be placed in the mutacc_root directory) and the job id for the slurm job of the case.
 2. Running  ```$ cg upload process-solved --days-ago <days>``` should log the same as above for all the cases that have been updated within the specified number of days.
 3. Running ```$ cg upload processed-solved``` should find the cases that have been processed by mutacc and import them to the mutacc database

**Outcome**

$ cg upload process-solved --days-ago 17
----------------- UPLOAD ----------------------
----------------- PROCESS-SOLVED ----------------
Start processing case casualgannet with mutacc
Extracting reads from case casualgannet
mutacc_auto.cli.root[144032] INFO Running mutacc_auto
mutacc_auto.cli.extract_command[144032] INFO All files are placed in /home/proj/stage/rare-disease/mutacc_root/temporaries/mutacc_auto/tmpksmqupbd
Submitted batch job 152953
Start processing case strongbison with mutacc
Extracting reads from case strongbison
mutacc_auto.cli.root[144037] INFO Running mutacc_auto
mutacc_auto.cli.extract_command[144037] INFO All files are placed in /home/proj/stage/rare-disease/mutacc_root/temporaries/mutacc_auto/tmpouyl3ltw
Submitted batch job 152954

$ cg upload process-solved --case-id onemite
----------------- UPLOAD ----------------------
----------------- PROCESS-SOLVED ----------------
Fetching case onemite
Start processing case onemite with mutacc
Extracting reads from case onemite
mutacc_auto.cli.root[144499] INFO Running mutacc_auto
mutacc_auto.cli.extract_command[144499] INFO All files are placed in /home/proj/stage/rare-disease/mutacc_root/temporaries/mutacc_auto/tmp3ri5yf5a
Submitted batch job 152955

$ cg upload processed-solved
----------------- UPLOAD ----------------------
----------------- PROCESSED-SOLVED ----------------
Uploading processed cases by mutacc to the mutacc database
mutacc_auto.cli.root[159975] INFO Running mutacc_auto
mutacc_auto.cli.import_command[159975] INFO importing /home/proj/stage/rare-disease/mutacc_root/imports/strongbison_import_mutacc.json
mutacc.cli.root[159979] INFO Running mutacc
mutacc.cli.database[159979] INFO Establishing connection with host …, on port …
mongo_adapter.client[159979] INFO Connecting to uri:mongodb://…:…@…:…
mongo_adapter.client[159979] INFO Connection established
mutacc.cli.importing[159979] INFO Inserting case from /home/proj/stage/rare-disease/mutacc_root/imports/strongbison_import_mutacc.json into mutaccDB
mutacc.mutaccDB.db_adapter[159979] WARNING No case with case_id strongbison
mutacc.mutaccDB.insert[159979] INFO Case added to mutaccDB
mutacc_auto.cli.import_command[159975] INFO importing /home/proj/stage/rare-disease/mutacc_root/imports/onemite_import_mutacc.json
mutacc.cli.root[159991] INFO Running mutacc
mutacc.cli.database[159991] INFO Establishing connection with host …, on port …
mongo_adapter.client[159991] INFO Connecting to uri:mongodb://…:…@…:…
mongo_adapter.client[159991] INFO Connection established
mutacc.cli.importing[159991] INFO Inserting case from /home/proj/stage/rare-disease/mutacc_root/imports/onemite_import_mutacc.json into mutaccDB
mutacc.mutaccDB.db_adapter[159991] WARNING No case with case_id onemite
mutacc.mutaccDB.insert[159991] INFO Case added to mutaccDB
mutacc_auto.cli.import_command[159975] INFO importing /home/proj/stage/rare-disease/mutacc_root/imports/casualgannet_import_mutacc.json
mutacc.cli.root[159998] INFO Running mutacc
mutacc.cli.database[159998] INFO Establishing connection with host …, on port …
mongo_adapter.client[159998] INFO Connecting to uri:mongodb://...:...@...:…
mongo_adapter.client[159998] INFO Connection established
mutacc.cli.importing[159998] INFO Inserting case from /home/proj/stage/rare-disease/mutacc_root/imports/casualgannet_import_mutacc.json into mutaccDB
mutacc.mutaccDB.db_adapter[159998] WARNING No case with case_id casualgannet
mutacc.mutaccDB.insert[159998] INFO Case added to mutaccDB

**Review:**
- [x] code approved by @patrikgrenfeldt 
- [x] tests executed by @adrosenbaum 
Thanks for filling in who performed the code review and the test!

This is a minor **version bump** because it adds a new feature in a backward compatible manner
